src/ArrayDataException.java

```diff
@@ -0,0 +1,5 @@
+public class ArrayDataException extends Exception {
```

Comment:
Extending from `Exception` is not really correct. Would be enough to extend from `RuntimeException`